File: src/entities/external/PodList.ts
export type PodList = {
  kind: string;
  apiVersion: string;
  metadata: { resourceVersion: string };
  items: Item[];
};

export type Item = {
  metadata: ItemMetadata;
  spec: any;
  status: Status;
};

export type ItemMetadata = {
  name: string;
  generateName: string;
  namespace: string;
  uid: string;
  resourceVersion: string;
  creationTimestamp: Date;
  labels: Labels;
  annotations: Annotations;
  ownerReferences: OwnerReference[];
  managedFields: any[];
};

export type Annotations = {
  "kubectl.kubernetes.io/default-container": string;
  "kubectl.kubernetes.io/default-logs-container": string;
  "prometheus.io/path": string;
  "prometheus.io/port": string;
  "prometheus.io/scrape": string;
  "sidecar.istio.io/status": string;
};

export type Labels = {
  app: string;
  "pod-template-hash": string;
  "security.istio.io/tlsMode": string;
  "service.istio.io/canonical-name": string;
  "service.istio.io/canonical-revision": string;
  version: string;
};

export type OwnerReference = {
  apiVersion: string;
  kind: string;
  name: string;
  uid: string;
  controller: boolean;
  blockOwnerDeletion: boolean;
};

export type Status = {
  phase: string;
  conditions: Condition[];
  hostIP: string;
  podIP: string;
  podIPs: { ip: string }[];
  startTime: Date;
  initContainerStatuses: any[];
  containerStatuses: any[];
  qosClass: string;
};

export type Condition = {
  type: string;
  status: string;
  lastProbeTime: null;
  lastTransitionTime: Date;
};

File: kmamiz_data_processor/src/data/pod_list.rs
use serde::{ Deserialize, Serialize };

#[derive(Debug, Serialize, Deserialize, Default)]
pub struct PodList {
    pub kind: String,
    #[serde(rename = "apiVersion")]
    pub api_version: String,
    pub items: Vec<Item>,
}

#[derive(Debug, Serialize, Deserialize, Default)]
pub struct Item {
    pub metadata: ItemMetadata,
}

#[derive(Debug, Serialize, Deserialize, Default)]
pub struct ItemMetadata {
    pub name: String,
    #[serde(rename = "generateName")]
    pub generate_name: String,
    pub namespace: String,
    pub uid: String,
    #[serde(rename = "resourceVersion")]
    pub resource_version: String,
    #[serde(rename = "creationTimestamp")]
    pub creation_timestamp: String,
    pub labels: Labels,
    pub annotations: Annotations,
}

#[derive(Debug, Serialize, Deserialize, Default)]
pub struct Annotations {
    #[serde(rename = "kubectl.kubernetes.io/default-container")]
    pub kubectl_kubernetes_io_default_container: String,
    #[serde(rename = "kubectl.kubernetes.io/default-logs-container")]
    pub kubectl_kubernetes_io_default_logs_container: String,
    #[serde(rename = "prometheus.io/path")]
    pub prometheus_io_path: String,
    #[serde(rename = "prometheus.io/port")]
    pub prometheus_io_port: String,
    #[serde(rename = "prometheus.io/scrape")]
    pub prometheus_io_scrape: String,
    #[serde(rename = "sidecar.istio.io/status")]
    pub sidecar_istio_io_status: String,
}

#[derive(Debug, Serialize, Deserialize, Default)]
pub struct Labels {
    pub app: String,
    #[serde(rename = "pod-template-hash")]
    pub pod_template_hash: String,
    #[serde(rename = "security.istio.io/tlsMode")]
    pub security_istio_io_tls_mode: String,
    #[serde(rename = "service.istio.io/canonical-name")]
    pub service_istio_io_canonical_name: String,
    #[serde(rename = "service.istio.io/canonical-revision")]
    pub service_istio_io_canonical_revision: String,
    pub version: String,
}